### PR TITLE
Rename `sync-protocol.md` to `light-client-sync.md` and executable Altair validator guide

### PR DESCRIPTION
**Pull request by hwwhww**
1. When `sync-protocol.md` was in `lightclient` folder, the name `sync-protocol.md` made sense. But since now the specs are in `altair` folder, it should be renamed to `light-client-sync.md`, which similar to `phase1/light-client-sync.md`.
2. Make Altair validator guide pass the linter checks.

### CODE REVIEW

File: specs/altair/validator.md [hunk 1]
```diff
@@ -137,7 +146,9 @@ This function is a predicate indicating the presence or absence of the validator
 ```python
 def compute_sync_committee_period(epoch: Epoch) -> uint64:
     return epoch // EPOCHS_PER_SYNC_COMMITTEE_PERIOD
+```
```
Comment (ralexstokes) [inline specs/altair/validator.md:149]: i guess each function needs to be in its own code block (?)

File: specs/altair/validator.md [hunk 2]
```diff
@@ -280,9 +291,10 @@ This function returns multiple subnets if a given validator index is included mu
 ```python
 def compute_subnets_for_sync_committee(state: BeaconState, validator_index: ValidatorIndex) -> Sequence[uint64]:
     target_pubkey = state.validators[validator_index].pubkey
-    sync_committee_indices = [index for index, pubkey in enumerate(state.current_sync_committee.pubkeys) if pubkey == target_pubkey]
+    sync_committee_indices = [index for index, pubkey in enumerate(state.current_sync_committee.pubkeys)
+                              if pubkey == target_pubkey]
     return [
-        uint64(index // (SYNC_COMMITEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT))
+        uint64(index // (SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT))
```
Comment (ralexstokes) [inline specs/altair/validator.md:297]: ah! thanks, i always misspell committee

File: specs/altair/validator.md [hunk 3]
```diff
@@ -125,6 +126,14 @@ class SignedContributionAndProof(Container):
     signature: BLSSignature
 ```
 
+### `SyncCommitteeSigningData`
+
+```python
+class SyncCommitteeSigningData(Container):
+    slot: Slot
+    subcommittee_index: uint64
+```
```
Comment (ralexstokes) [inline specs/altair/validator.md:135]: this works for me -- i'll note there are some `Container` definitions in the phase 0 validator guide that are scattered throughout the text and not collected at the top -- something we can eventually fix

**Review by ralexstokes — APPROVED** [review]
looks good to me! i was planning on adding some validator tests so thanks for putting the validator.md file into the "executable framework"

File: specs/altair/validator.md (hunk 3 above)
Comment (ralexstokes) [reply specs/altair/validator.md:135]: https://github.com/ethereum/eth2.0-specs/pull/2251

**Comment by djrtwo** [comment]
I think @hwwhww is going to do a bit more reorganization before final review. changing this to "DO NOT MERGE"

**Comment by hwwhww** [comment]
Thanks for the review @ralexstokes! 

I'm writing light client P2P protocol. Talked with @djrtwo and it may be better to have:
```
specs/altair/light-client/README.md
specs/altair/light-client/sync-protocol.md
specs/altair/light-client/p2p-interface.md
```

So closing this PR for now for cleaner commit history. will make another short PR for the validator.md minor fixes.

**Pull request closed without merging**